### DISCUSSION TITLE
Set pool sizes and IDLE timeout

### DISCUSSION THREAD

**Discussion by eduhAgostinho** [ebean-orm/ebean#3316] [General]
How to set the pool maximum size and IDLE timeout correctly with Ebean? I've tried changing the env variables adding the following code but none of them worked:
```
ebean.datasource.pool.maxSize=${EBEANDATASOURCEPOOLMAXSIZE} 
spring.datasource.hikari.maximum-pool-size=${EBEANDATASOURCEPOOLMAXSIZE} 
spring.datasource.hikari.idleTimeout=${EBEANDATASOURCEIDLETIMETOU} 
```

**Comment by rbygrave** [comment]
Sorry, I think you need to clarify your question.  

It looks like you are using hikari connection pool ... in which case this isn't really a ebean question?
It also looks like the hikari connection pool is being configured via spring? In which case the hikari connection pool is being configured by spring? (but there isn't any spring wiring code here like `@Configuration` `@Bean` so perhaps its spring boot auto configuration of Hikari).

Do you want to clarify the question a bit?

**Comment by eduhAgostinho** [comment]
Basically my question is, how can I set a maximum pool size using Ebean? Is there a way to do that? I included the Hikari configuration hoping that somehow Hikari was integrated with Ebean, which it seems it is not.

**Reply by rbygrave** [reply]
We can specify any Datasource to use or use ebean-datasource, and when we use ebean-datasource we could specify maxConnections like:

Assuming we are using the default name of the ebean database (which is `db`) then the name of the datasource will also use that same name (so `db`) then the configuration properties for the datasource start with `datasource.db.` 

So configuration like:

```
datasource.db.username=sa
datasource.db.password=
datasource.db.url=jdbc:h2:mem:testsMem;DB_CLOSE_ON_EXIT=FALSE;NON_KEYWORDS=KEY,VALUE
datasource.db.poolListener=org.tests.basic.MyTestDataSourcePoolListener
#datasource.db.minConnections=1
#datasource.db.maxConnections=25
#datasource.db.capturestacktrace=true
#datasource.db.maxStackTraceSize=20
#datasource.db.driver=org.h2.Driver
#datasource.db.heartbeatsql=select 1
#datasource.db.isolationlevel=read_committed
#datasource.db.customProperties=IGNORECASE=TRUE;MODE=Oracle;
``` 

Max Connections would be like:
```
datasource.db.maxConnections=200
```

Max idle time is called "maxInactiveTimeSecs" so
```
datasource.db.maxInactiveTimeSecs=300
```

The code that is loading the configuration (if you are looking where to debug) is at: https://github.com/ebean-orm/ebean-datasource/blob/master/ebean-datasource-api/src/main/java/io/ebean/datasource/DataSourceConfig.java#L707-L757

**Reply by rbygrave** [reply]
Also just to say, an application can be multiple ebean databases and in that case they are given names and most folks would not use the default name of `db` in that case.  

e.g. If the database name was `foo` then max connections would be:

```
datasource.foo.maxConnections=200
```

-------

Also, for optimal performance ebean will run with a second Datasource optimised for read only use (readOnly=true, autoCommit=true) and often this is targeting a read replica (e.g. AWS Aurora read replica). So with ebean for this case we can set the readOnlyDatasource ... or when using ebean-datasource set the `readOnlyUrl` like: https://github.com/ebean-orm/ebean-datasource/blob/master/ebean-datasource-api/src/main/java/io/ebean/datasource/DataSourceConfig.java#L719
